geomesa-fs/geomesa-fs-tools/src/main/scala/org/locationtech/geomesa/fs/tools/compact/CompactCommand.scala

```diff
@@ -0,0 +1,90 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.fs.tools.compact
+
+import java.io.File
+
+import com.beust.jcommander.{Parameter, ParameterException, Parameters}
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.fs.Path
+import org.locationtech.geomesa.fs.FileSystemDataStore
+import org.locationtech.geomesa.fs.tools.ingest.TempDirParam
+import org.locationtech.geomesa.fs.tools.{FsDataStoreCommand, FsParams, PartitionParam}
+import org.locationtech.geomesa.tools.ingest.AbstractIngest
+import org.locationtech.geomesa.tools.ingest.AbstractIngest.PrintProgress
+import org.locationtech.geomesa.tools.{Command, RequiredTypeNameParam}
+import org.locationtech.geomesa.utils.classpath.ClassPathUtils
+import org.locationtech.geomesa.utils.text.TextTools
+
+import scala.collection.JavaConversions._
+
+class CompactCommand extends FsDataStoreCommand with LazyLogging {
+
+  override val name: String = "compact"
+  override val params = new CompactParams
+
+  override def execute(): Unit = {
+    withDataStore(compact)
+  }
+
+  val libjarsFile: String = "org/locationtech/geomesa/fs/tools/ingest-libjars.list"
+
+  def libjarsPaths: Iterator[() => Seq[File]] = Iterator(
+    () => ClassPathUtils.getJarsFromEnvironment("GEOMESA_FS_HOME"),
+    () => ClassPathUtils.getJarsFromClasspath(classOf[FileSystemDataStore])
+  )
+
+  def compact(ds: FileSystemDataStore): Unit = {
+    Command.user.info(s"Beginning Compaction Process...updating metadata")
+
+    ds.storage.updateMetadata(params.featureName)
+    Command.user.info(s"Metadata update complete")
+
+    val m = ds.storage.getMetadata(params.featureName)
+    val allPartitions = m.getPartitions
+    val toCompact: Seq[String] = if (params.partitions.nonEmpty) {
+      params.partitions.filterNot(allPartitions.contains).headOption.foreach { p =>
+        throw new ParameterException(s"Partition $p cannot be found in metadata")
+      }
+      params.partitions
+    } else {
+      allPartitions
+    }
+    Command.user.info(s"Compacting ${toCompact.size} partitions")
+
+    params.runMode match {
```

Comment:
would be good to validate the runMode before this so there's a clear error instead of a match exception